controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -52,54 +52,42 @@ beerocks::eChannelScanStatusCode dynamic_channel_selection_task::dcs_request_sca
     }
-    request->scan_params().dwell_time_ms = dwell_time_msec;
 
-    // Get channel pool;
-    // if pool is set to "all channel": use the supported channels instead.
-    // if the costum channel pool is empty or too big: fail scan and return error accordingly.
+    //get current channel pool from DB
     auto &curr_channel_pool = database.get_channel_scan_pool(m_radio_mac, m_single_scan);
-    if (curr_channel_pool.size() == 1 && (*curr_channel_pool.begin()) == 0) {
-        LOG(DEBUG) << "Using all supported channels for channel scan on mac=" << m_radio_mac;
-        auto supported_channels = database.get_hostap_supported_channels(radio_mac_str);
-        request->scan_params().channel_pool_size = supported_channels.size();
-        std::transform(supported_channels.begin(), supported_channels.end(),
-                       request->scan_params().channel_pool,
-                       [](beerocks::message::sWifiChannel &c) -> uint8_t { return c.channel; });
-    } else {
-        LOG(DEBUG) << "Using custom set channels for channel scan on mac=" << m_radio_mac;
-        if (curr_channel_pool.empty()) {
-            LOG(ERROR) << "empty channel pool is not supported. please set channel pool for mac="
-                       << m_radio_mac;
-            return beerocks::eChannelScanStatusCode::INVALID_PARAMS;
-        }
-        if (curr_channel_pool.size() > beerocks::message::SUPPORTED_CHANNELS_LENGTH) {
-            LOG(ERROR) << "channel_pool is too big [" << int(curr_channel_pool.size())
-                       << "] on mac=" << m_radio_mac;
-            return beerocks::eChannelScanStatusCode::POOL_TOO_BIG;
-        }
-        request->scan_params().channel_pool_size = curr_channel_pool.size();
-        std::copy(curr_channel_pool.begin(), curr_channel_pool.end(),
-                  request->scan_params().channel_pool);
+    if (curr_channel_pool.empty()) {
+        LOG(ERROR) << "empty channel pool is not supported. please set channel pool for mac="
+                   << m_radio_mac;
+        return beerocks::eChannelScanStatusCode::INVALID_PARAMS;
     }
 
+    if (curr_channel_pool.size() > beerocks::message::SUPPORTED_CHANNELS_LENGTH) {
+        LOG(ERROR) << "channel_pool is too big [" << int(curr_channel_pool.size())
+                   << "] on mac=" << m_radio_mac;
+        return beerocks::eChannelScanStatusCode::POOL_TOO_BIG;
+    }
+
+    request->scan_params().radio_mac         = m_radio_mac;
+    request->scan_params().dwell_time_ms     = dwell_time_msec;
+    request->scan_params().channel_pool_size = curr_channel_pool.size();
+    std::copy(curr_channel_pool.begin(), curr_channel_pool.end(),
+              request->scan_params().channel_pool);
+
     // get the parent node to send the CMDU to the agent
-    auto ire = database.get_node_parent_ire(radio_mac_str);
+    auto radio_mac_str = tlvf::mac_to_string(m_radio_mac);
+    auto ire           = database.get_node_parent_ire(radio_mac_str);
```

Comment:
need to add a check here:
if we have for some reason `get_node_parent_ire` return an empty string - we don't check this and perform `send_cmdu_to_agent_ with empty std::string

Reply:
Added check, in case `ire` returns as an empty string the DCS will reach the `Abort` state with an `INTERNAL_FAILURE` status